### ISSUE TITLE
Grid editor row column widths are not updated when columns are resized

### ISSUE THREAD

**Issue by vaadin-bot** [vaadin/framework#7276]
**Originally by @elmot**

___
Steps to reproduce:
 - open http://demo.vaadin.com/sampler-7.6/#ui/grids-and-trees/grid/item-editor
 - click any column to activate row editor
 - resize grid
 - observe funny open popup dragging, jumping and even collapsing in some circumstances

___
 Imported from https://dev.vaadin.com/ issue #19212

**Comment by leech74** [comment]
Hi, please fix this. It is still happening in 8.1.3.

**Comment by FollowSteph** [comment]
Please consider fixing this issue. It's been reported for 2.5 years...

**Comment by TatuLund** [comment]
At the moment Grid's editor calculates column widths only when opening.

There is workaround for this

```
		grid.addColumnResizeListener(event -> {
			if (grid.getEditor().isOpen()) {
				grid.getEditor().cancel();
				grid.getEditor().editRow(lastEditedRow);
			}
		});

```

Unfortunately you get only the bean from Grid Editor opening event

		grid.getEditor().addOpenListener(event -> {
			event.getBean();
                        lastEditedRow = resolveRowNumber(event.getBean());
		});

In [GridFastNavigation add-on](https://vaadin.com/directory/component/gridfastnavigation-add-on) there is alternative open event, that gives also the row. I tried the workaround with it and it seemed to work reasonably well.

**Comment by pwilkin** [comment]
This is not a workaround. Cancelling the editor when it's in buffered mode discards all changes. Using this method, a user resizing a column would discard all of their edited changes.

Please consider making a proper fix for this.

**Comment by TatuLund** [comment]
There is proper fix in GridFastNavigation add-on version 2.3.5 (however it is usable only in un-buffered mode), I am also considering adding simple extension to add-on that can be used with buffered editor as well.

https://vaadin.com/directory/component/gridfastnavigation-add-on

**Comment by TatuLund** [comment]
I published also standalone extension with the fix, that can be used with both buffered and un-buffered editor, when GridFastNavigation is not needed.

https://vaadin.com/directory/component/grideditorcolumnfix-add-on
